### PR TITLE
feat: tooltip allow for tooltips with variables

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Start to make the tooltips allow for having passed props.
- Make us of type `ReactNode | ((params: any) => ReactNode)` (@alfetopito )
- Added helper function`renderTooltip` to 'legacy/components/Tooltip'
- Minor styling

# Todo:  (need help on this one)
- Properly handle the time unit + show days/hours/minutes in the Total Duration tooltip.

Screenshot 2023-06-06 at 17 26 17

### CODE REVIEW

File: src/modules/trade/pure/TradeWidgetField/index.tsx [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
-  hint?: JSX.Element | string
+  tooltip?: React.ReactNode | ((params: any) => React.ReactNode)
   error?: TradeWidgetFieldError
   className?: string
 }
 
 export function TradeWidgetField(props: TradeWidgetFieldProps) {
-  const { className, children, label, hint, error } = props
+  const { className, children, label, tooltip, error } = props
+  const tooltipElement = renderTooltip(tooltip, props)
```
Comment (alfetopito) [inline src/modules/trade/pure/TradeWidgetField/index.tsx:22]: Isn't this the only place where you'll ever have maybe a fn tooltip?
Everywhere else can just get the tooltip directly, no?

File: src/modules/twap/containers/TwapFormWidget/index.tsx [hunk 2]
```diff
@@ -101,10 +103,14 @@ export function TwapFormWidget() {
           deadline={deadlineState}
           items={orderDeadlines}
           setDeadline={(value) => updateSettingsState(value)}
-          labels={LABELS_TOOLTIPS.totalDuration}
+          label={LABELS_TOOLTIPS.totalDuration.label}
+          tooltip={renderTooltip(LABELS_TOOLTIPS.totalDuration.tooltip, {
+            parts: numberOfPartsValue,
+            partDuration: timeInterval,
+          })}
```
Comment (alfetopito) [inline src/modules/twap/containers/TwapFormWidget/index.tsx:110]: Ok, that's the second place where you explicitly have a tooltip fn, but then, why not just call the fn here?

**Comment by alfetopito** [comment]
One issue with the total duration for example (which is not directly touched here but I'm raising anyway) is the fact that the selector changes position depending on the time selected

https://github.com/cowprotocol/cowswap/assets/43217/a7a08994-a4f9-4eb7-bd18-a82ddb0aebc9

**Review by elena-zh — COMMENTED** [review]
UI looks good to me. 
However, numbers are not (todo).  :)
Also, warnings are not yet implemented for the Min part. However, I'd better ask: will we have a 'special' text in the tooltip when a part's time is less than 5 min? 

Also, might be not related to the current PR, but `buyAmount.label` label does not look good to me here: 
image

**Comment by elena-zh** [comment]
Since my questions were moved to #2600, changes LGTM!

**Pull request merged**